tests/profiling/conftest.py

```diff
@@ -0,0 +1,117 @@
+"""Some pytest configuration specific to the profiling test suite."""
+
+from importlib import resources
+
+import numpy as np
+import pytest
+import xarray
+
+
+@pytest.fixture(scope="session")
```

Comment:
Maybe the `pmodel_profile_data` fixture can be moved to `test_profiling_pmodel` and the "session" scope can be changed to "module" so that this fixture will be destroyed after the `pmodel` profiling test, preventing both `pmodel_profile_data` and `splash_profile_data` occupying the RAM together.

Reply:
@tztsai That's a good point - it is tricky though because I've separated tests into modules, so data fixtures are  shared via conftest.py and the execution order is just set by the test file name.

I think the cleanest way to do this is to separate the profiling tests into packages, each of which has its own `conftest.py` with package scoped data fixtures. That should create stable groups of tests that use the same data and should also scope the fixture creation and tear down correctly.

Although - when I actually _do_ that I don't seem to see a shift in peak memory usage.